test/Validate.SSA.cpp

```diff
@@ -1172,6 +1173,67 @@ TEST_F(ValidateSSA, PhiUseDoesntDominateUseOfPhiOperandUsedBeforeDefinitionBad)
     MatchesRegex("ID .\\[inew\\] has not been defined"));
 }
 
+TEST_F(ValidateSSA, PhiUseMayComeFromNonDominatingBlockGood) {
+  string str = kHeader
+      + "OpName %if_true \"if_true\"\n"
+      + "OpName %exit \"exit\"\n"
+      + "OpName %copy \"copy\"\n"
+      + kBasicTypes +
+               R"(
+%func        = OpFunction %voidt None %vfunct
+%entry       = OpLabel
+               OpBranchConditional %false %if_true %exit
+
+%if_true     = OpLabel
+%copy        = OpCopyObject %boolt %false
+               OpBranch %exit
+
+; The use of %copy here is ok, even though it was defined
+; in a block that does not dominate %exit.  That's the point
+; of an OpPhi.
+%exit        = OpLabel
+%value       = OpPhi %boolt %false %entry %copy %if_true
+               OpReturn
+               OpFunctionEnd
+)";
+
+  CompileSuccessfully(str);
+  ASSERT_EQ(SPV_SUCCESS, ValidateInstructions()) << getDiagnosticString();
+}
+
+TEST_F(ValidateSSA, DISABLED_PhiVariableDefMustComeFromBlockDominatingThePredecessorBad) {
```

Comment:
Hmm. I am sure this was the intention but I don't see a reference to this rule anywhere in the spec. Here is what the instruction definition says about the `Parent` `Variable` pair:

```
Operands are a sequence of pairs: (/Variable 1/, /Parent 1/ block), (/Variable
2/, /Parent 2/ block),... Each /Parent i/ block is the label of an immediate
predecessor in the CFG of the current block. A /Parent i/ block must not appear
more than once in the operand sequence. All /Variables/ must have a type
matching /Result Type/.
```